src/Http/Routing.Abstractions/src/RouteData.cs

```diff
@@ -138,14 +141,16 @@ public RouteValueDictionary Values
         /// <see cref="DataTokens"/> will not be changed.
         /// </param>
         /// <returns>A <see cref="RouteDataSnapshot"/> that captures the current state.</returns>
-        public RouteDataSnapshot PushState(IRouter router, RouteValueDictionary values, RouteValueDictionary dataTokens)
+        public RouteDataSnapshot PushState(IRouter? router, RouteValueDictionary? values, RouteValueDictionary? dataTokens)
```

Comment:
I'm not sure I don't think it make sense for these to receive null values. I would be in favor of making them non-nullable to match our expectations.

Reply:
https://github.com/dotnet/aspnetcore/blob/85e2147ff037cc9950249fbb9d095ad47ec4184a/src/Http/Routing/src/RouteCollection.cs#L61 I kinda don't want to touch it when it says perf optimization.